### ISSUE TITLE
Routing configuration should be reworked to avoid parameter order hardcoding

### ISSUE THREAD

**Issue by zecrazytux** [viking-gps/viking#37]
The routing module, thanks to the XML configuration module, can use other/custom routing API. However, the coordinates are hardcoded in this order: latitude,longitude. This prevents from using APIs that take as arguments coordinates in longitude,latitude format, which is the case for example of brouter.

I did not try to propose some code as a pull request as I failed to rebuild viking-gps from debian sources (something related to linking a static lib and a dynamic one, with an advice to rebuild with -fPIC... I'm not competent to deal with it !) with minor modification (simply tried to change lat,lon by lon,lat).
however, I was able to use brouter by using the following routing.xml file
```
<objects>
  <object class="VikRoutingWebEngine">
    <property name="id">brouter</property>
    <property name="label">brouter</property>
    <property name="format">gpx</property>
    <property name="url-base">http://h2096617.stratoserver.net:443/brouter?nogos=&amp;profile=hiking-beta&amp;alternativeidx=0&amp;format=gpx</property>
    <property name="url-start-ll">&amp;lonlats=%s,%s</property>
    <property name="url-stop-ll">|%s,%s</property>
    <property name="url-via-ll"></property>
  </object>
</objects>
```

and by reordering lon/lat by using a MITM "attack".

brouter with the hiking-beta profile is very efficient for creating tracks for hiking, trail running, mountain cycling... It'ld be great to add support for it !

thanks !

**Comment by rnorris** [comment]
Agreed, the format should be more flexible.

ATM I think the most likely (least effort to implement & therefore get done) approach will be to add an extra property along the lines of:

    <property name="ll-order-lat-first">FALSE</property>

To switch the order to be lon, lat.

Thus this would keep any backwards compatibly as the default would be TRUE to use lat, lon.

Ultimately a more flexible tokenizer approach would be nice and/or a smarter way to select sub-items (e.g. being able to list/select the routing profile type [cycle/car/foot/etc...] - without having to specify the whole object class common properties each time), but unlikely to get done in the near future.

**Comment by rnorris** [comment]
Addressed with #39

Add since you can also change the default ordering via the "printf positional arguments" technique '%n$' for the Nth argument - there is no need for an additional parameter reordering methods.
